test/e2e/proxy_test.go

```diff
@@ -253,6 +256,35 @@ func TestProxyFlow(t *testing.T) {
 				assert.Equal(t, expectedApp.Spec.DisplayName, createdApp.Spec.DisplayName)
 			}) // end of successful workspace context request
 
+			t.Run("successful workspace context request with proxy plugin", func(t *testing.T) {
+				// we are going to repurpose a well known, always running route as a proxy plugin to contact through the registration service
+				CreateProxyPluginWithCleanup(t, hostAwait, "openshift-console", "openshift-console", "console")
+				VerifyProxyPlugin(t, hostAwait, "openshift-console")
+				proxyPluginWorkspaceURL := hostAwait.PluginProxyURLWithWorkspaceContext("openshift-console", user.compliantUsername)
+				client := http.Client{
+					Timeout: 30 * time.Second,
+					Transport: &http.Transport{
+						TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec
+					},
+				}
+				request, err := http.NewRequest("GET", proxyPluginWorkspaceURL, nil)
+				require.NoError(t, err)
+
+				err = FromString(t, user.token)
+				t.Logf("GGM from string for user token len %d returned error %#v token %s", len(user.token), err, user.token)
+
+				request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", user.token))
```

Comment:
Based on the debug info you posted here: https://github.com/codeready-toolchain/registration-service/pull/311#issuecomment-1469943416

I can tell that the token you log here is the user's sso token as expected. However the token which is logged in the proxy is the member operator service account token which is not expected.

I have no idea why it's so and why the request contains the wrong Authorization header. Looks like something wrong with the client initialization but I don't see it... Everything looks good. But just to make sure there is no other headers added somewhere else let's use Set instead of Add.

```suggestion
				request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", user.token))
```